### ISSUE TITLE
Unable to boot using builds made in the last week or so

### ISSUE THREAD

**Issue by trev-dev** [NixOS/nixpkgs#255053]
### Describe the bug
I can't tell if my hard drive is starting to tilt or something has changed upstream and is now breaking my things.

TL:DR; nixos rebuilds on/after 2023-09-11 result in my not being able to boot my machine after decrypting my Luks partition.  I get the following messages after entering the correct password after about 45 seconds:

```
Unable to change power state from D3cold to D0
Buffer I/O error on dev nvme0n1p2 logical block 0 async page read
(The last message for blocks 1-11.  nvme0n1p2 is my root partition.  We are booting from p1)
Device /dev/disk/by-uuid/<big UUID> is not a valid Luks device.
```

If I roll back to my build from 2023-09-03, I can boot up fine, consistently, every time.

### Steps To Reproduce
1. I have no idea.

### Expected behavior
To be able to update my nix system and still be able to boot

### Screenshots
![ruh-roh](https://github.com/NixOS/nixpkgs/assets/28788713/f05cc937-5e0f-47ad-9104-947ffccf379a)

### Additional context
My smartmontools seems to be happy:
```
=== START OF INFORMATION SECTION ===
Model Number:                       THNSN5512GPUK NVMe TOSHIBA 512GB
Serial Number:                      27GB700EKSHU
Firmware Version:                   5KDA4103
PCI Vendor/Subsystem ID:            0x1179
IEEE OUI Identifier:                0x00080d
Controller ID:                      0
NVMe Version:                       <1.2
Number of Namespaces:               1
Namespace 1 Size/Capacity:          512,110,190,592 [512 GB]
Namespace 1 Formatted LBA Size:     512
Namespace 1 IEEE EUI-64:            00080d 03000f0681
Local Time is:                      Wed Sep 13 19:37:45 2023 PDT
Firmware Updates (0x02):            1 Slot
Optional Admin Commands (0x0017):   Security Format Frmw_DL Self_Test
Optional NVM Commands (0x001e):     Wr_Unc DS_Mngmt Wr_Zero Sav/Sel_Feat
Log Page Attributes (0x02):         Cmd_Eff_Lg
Warning  Comp. Temp. Threshold:     78 Celsius
Critical Comp. Temp. Threshold:     82 Celsius

Supported Power States
St Op     Max   Active     Idle   RL RT WL WT  Ent_Lat  Ex_Lat
 0 +     6.00W       -        -    0  0  0  0        0       0
 1 +     2.40W       -        -    1  1  1  1        0       0
 2 +     1.90W       -        -    2  2  2  2        0       0
 3 -   0.0120W       -        -    3  3  3  3     5000   25000
 4 -   0.0060W       -        -    4  4  4  4   100000   70000

Supported LBA Sizes (NSID 0x1)
Id Fmt  Data  Metadt  Rel_Perf
 0 +     512       0         2
 1 -    4096       0         1

=== START OF SMART DATA SECTION ===
SMART overall-health self-assessment test result: PASSED

SMART/Health Information (NVMe Log 0x02)
Critical Warning:                   0x00
Temperature:                        38 Celsius
Available Spare:                    100%
Available Spare Threshold:          50%
Percentage Used:                    36%
Data Units Read:                    49,491,649 [25.3 TB]
Data Units Written:                 91,462,650 [46.8 TB]
Host Read Commands:                 653,712,905
Host Write Commands:                1,606,801,024
Controller Busy Time:               11,536
Power Cycles:                       4,208
Power On Hours:                     28,242
Unsafe Shutdowns:                   413
Media and Data Integrity Errors:    0
Error Information Log Entries:      0
Warning  Comp. Temperature Time:    0
Critical Comp. Temperature Time:    0
Temperature Sensor 1:               38 Celsius

Error Information (NVMe Log 0x01, 16 of 128 entries)
No Errors Logged
```

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.45, NixOS, 23.05 (Stoat), 23.05.2835.b30c68669df7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.5`
 - channels(root): `"nixos-23.05"`
 - channels(trev): `"home-manager-23.05.tar.gz, unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by fabianhjr** [comment]
I briefly experienced something similar on the 6.5.2 kernel when I crashed due to OOM kills; attempting to boot with 6.5.1 was enough to recover if you have access to such a build (and haven't experienced that again since)

**Comment by fabianhjr** [comment]
I experienced something similar on the 6.5.2 kernel, which would coincide with the 6.1.52 release from those dates.

(Though in my case there was a previous system crash / hard reset due to OOM kills)

Could you try the recent kernel updates? (6.1.53, currently in a merge request over at https://github.com/NixOS/nixpkgs/pull/254977)

**Comment by RaitoBezarius** [comment]
Reported by @lf- also AFAIK

**Comment by lf-** [comment]
Yes, this is a duplicate of https://github.com/NixOS/nixpkgs/issues/253418 because it's the same computer as mine. Try reverting your kernel to one of the working versions listed on there :) we should possibly do what opensuse did and revert the bad commit in a patch (as linked on bugzilla).
